examples/rest-api-client-dts/index.test-d.ts

```diff
@@ -43,3 +45,10 @@ export async function test() {
     repo: "javascript-plugin-architecture-with-typescript-definitions",
   });
 }
+
+export async function testInstanceType(client: RestApiClient) {
+  client.request("GET /repos/{owner}/{repo}", {
+    owner: "gr2m",
+    repo: "javascript-plugin-architecture-with-typescript-definitions",
+  });
+}
```

Comment:
@JoshuaKGoldberg Is there a way to make the above possible?

The only way I could figure out is to export both a `const` and a `type` like this:

```ts
export const RestApiClient: ExtendBaseWith<
  Base,
  {
    defaults: {
      userAgent: string;
    };
    plugins: [typeof requestPlugin];
  }
>;

export type RestApiClient = ExtendBaseWith<
  Base,
  {
    defaults: {
      userAgent: string;
    };
    plugins: [typeof requestPlugin];
  }
>;
```

Do you know if there is a way to combine both into one?

Reply:
Aha, yes: `export type RestApiClient = typeof RestApiClient;`, as you'd had in other places.

Reply:
I was worried at first about whether this would work (what a weird line!) -- but `typeof` dips into the _values_ space of names rather than _types_.